machine/v1beta1/types_azureprovider.go

```diff
@@ -437,11 +489,71 @@ type DiskEncryptionSetParameters struct {
 // SecurityProfile specifies the Security profile settings for a
 // virtual machine or virtual machine scale set.
 type SecurityProfile struct {
-	// This field indicates whether Host Encryption should be enabled
-	// or disabled for a virtual machine or virtual machine scale
-	// set. Default is disabled.
+	// encryptionAtHost indicates whether Host Encryption should be enabled or disabled for a virtual
+	// machine or virtual machine scale set.
+	// This should be disabled when SecurityEncryptionType is set to DiskWithVMGuestState.
+	// Default is disabled.
 	// +optional
 	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
+	// settings specify the security type and the UEFI settings of the virtual machine. This field can
+	// be set for Confidential VMs and Trusted Launch for VMs.
+	// +optional
+	Settings SecuritySettings `json:"settings,omitempty"`
+}
+
+// SecuritySettings define the security type and the UEFI settings of the virtual machine.
+// +union
+type SecuritySettings struct {
+	// securityType specifies the SecurityType of the virtual machine. It has to be set to any specified value to
+	// enable UEFISettings. The default behavior is: UEFISettings will not be enabled unless this property is set.
+	// +kubebuilder:validation:Enum=ConfidentialVM;TrustedLaunch
+	// +unionDiscriminator
+	// +required
+	SecurityType SecurityTypes `json:"securityType,omitempty"`
+	// confidentialVM specifies the security configuration of the virtual machine.
+	// For more information regarding Confidential VMs, please refer to:
+	// https://learn.microsoft.com/azure/confidential-computing/confidential-vm-overview
+	// +optional
+	ConfidentialVM *ConfidentialVM `json:"confidentialVM,omitempty"`
+	// trustedLaunch specifies the security configuration of the virtual machine.
+	// For more information regarding TrustedLaunch for VMs, please refer to:
+	// https://learn.microsoft.com/azure/virtual-machines/trusted-launch
+	// +optional
+	TrustedLaunch *TrustedLaunch `json:"trustedLaunch,omitempty"`
+}
+
+// ConfidentialVM defines the UEFI settings for the virtual machine.
+type ConfidentialVM struct {
+	// uefiSettings specifies the security settings like secure boot and vTPM used while creating the virtual machine.
+	// +required
```

Comment:
Should be `+kubebuilder:validation:Required`

Reply:
Nice catch, thank you, updated.